main.tf

```
@@ -46,12 +46,17 @@ resource "azurerm_eventgrid_system_topic_event_subscription" "this" {
 }
 
 resource "azurerm_monitor_diagnostic_setting" "this" {
-  name                       = "failure-logs"
-  target_resource_id         = azurerm_eventgrid_system_topic.this.id
-  log_analytics_workspace_id = var.log_analytics_workspace_id
+  name                           = "failure-logs"
```

Update with the new variable created for the diagnostic setting name.

```suggestion
  name                           = var.diagnostic_setting_name
```